### PR TITLE
feat: build showcase container image

### LINKED ISSUE #23: Showcase: build docker image and release it to the Container registry
The Timefold Container Registry ( https://github.com/orgs/TimefoldAI/packages ) should contain a docker image with the latest stable showcase so users can easily try it out.

That showcase container image should be able to run all the use-case quickstarts (preferably without having to install their container images).

### PR DESCRIPTION
**Pull request by rsynek**
TODO:

- [ ] reduce the image size (switch to alpine, explore jLink and sharing dependencies between QS)
- [ ] clean up the PR

### CODE REVIEW

**Comment by rsynek** [comment]
As I don't get to reducing the image size soon enough, I propose merging what we already have as the first step.

File: build/quickstarts-showcase/pom.xml [hunk 1]
```diff
@@ -167,6 +167,65 @@
...
+              <descriptors>
+                <descriptor>src/main/assembly/assembly-timefold-quickstarts.xml</descriptor>
+              </descriptors>
+              <finalName>container-content</finalName>
+              <appendAssemblyId>false</appendAssemblyId>
+            </configuration>
+          </plugin>
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>exec-maven-plugin</artifactId>
```
Comment (ge0ffrey) [inline build/quickstarts-showcase/pom.xml:203]: Is there a specialized docker maven plugin (the jib plugin maybe?) who can reduce the boilerplate in this exec approach, and avoid relying on installed programs in the environment?

File: build/quickstarts-showcase/src/main/assembly/scripts/runQuickstarts.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+# Change directory to the directory of the script
+cd "$(dirname $0)" || exit
+
+jvmOptions="-Xms128m -Xmx512m -Dstartup-open-browser=true"
+mainJar="quickstarts/binaries/quarkus-app/quarkus-run.jar"
+
+echo "Usage: ./runQuickstarts.sh"
+echo "Notes:"
+echo "- Java 11 or higher must be installed. Get the latest OpenJDK from (https://adoptium.net/)."
```
Comment (ge0ffrey) [inline build/quickstarts-showcase/src/main/assembly/scripts/runQuickstarts.sh]: We're minimum java 17

**Review by ge0ffrey — APPROVED** [review]

File: build/quickstarts-showcase/pom.xml (hunk 1 above)
Comment (ge0ffrey) [reply build/quickstarts-showcase/pom.xml:203]: Happy to push this to a later issue (let's create on then)

Comment (mswiderski) [reply build/quickstarts-showcase/pom.xml:203]: There is no need for any maven plug-in nor docker file. Just use jib quarkus extension and that’s it. Same as we do for nga and now also for employee scheduling in platform

File: build/quickstarts-showcase/src/main/assembly/scripts/runQuickstarts.sh (hunk 2 above)
Comment (rsynek) [reply build/quickstarts-showcase/src/main/assembly/scripts/runQuickstarts.sh]: The way we currently run the showcase, we don't even need the script. I will remove it.

File: build/quickstarts-showcase/pom.xml (hunk 1 above)
Comment (rsynek) [reply build/quickstarts-showcase/pom.xml:203]: The jib quarkus extension is a great thing, if the goal is to containerize a single quarkus app. Unfortunately, that's not what's happening here: there are 9 different applications in the container and one call the others.

To have all the content inside, we have to:

1. build the showcase app
2. grab the individual quickstarts and pack it with the showcase; that's what the assembly plugin does
3. put it all in a container image

The step 1. is when the maven quarkus plugin runs, including the jib extension.

After experimenting with two runs of the quarkus plugin, I also tried the jib maven plugin. Unfortunately, it does not allow just building the image locally; it always tries to push to some docker registry. What means you either have a local registry running or use the github registry directly. In my opinion, that is worse than having either `podman` or `docker` installed. Btw., you can switch the container runtime with a maven property.

Comment (mswiderski) [reply build/quickstarts-showcase/pom.xml:203]: just discussed that with @rsynek and `jib extension` does not fit here as this image will aggregate all the QuickStarts into single image. So, taking into account the need at the moment let's not overcomplicate it

**Review by mswiderski — APPROVED** [review]

**Pull request merged**